NewPass/PasswordGeneratorViewModel.swift

```diff
@@ -10,34 +10,34 @@ import Foundation
 import UIKit
 
 class PasswordGeneratorViewModel {
-    
+
     var passwordLength: Int
     // Initial password attributes match default attributes
     var passwordAttributes: [PasswordAttribute]
     var hasSelectedPasswordAttributes: Bool {
-        return passwordAttributes != []
+        return !passwordAttributes.isEmpty
     }
-    
+
     init(passwordLength: Int = Constants.DEFAULT_PASSWORD_LENGTH, passwordAttributes: [PasswordAttribute] = [.containsLowercaseLetters, .containsUppercaseLetters, .containsNumbers]) {
         self.passwordLength = passwordLength
         self.passwordAttributes = passwordAttributes
     }
-    
+
     func generateRandomPassword(length: Int) -> NSAttributedString {
         let randomPassword = Password.randomPassword(with: passwordAttributes, length: length)
-        
+
         return attributedPasswordString(from: randomPassword)
     }
-    
+
     private func attributedPasswordString(from passwordString: String) -> NSAttributedString {
         let symbolCharColor = [NSAttributedStringKey.foregroundColor: UIColor(red: 237/255, green: 117/255, blue: 99/255, alpha: 1)]
         let numberCharColor = [NSAttributedStringKey.foregroundColor: UIColor(red: 74/255, green: 196/255, blue: 230/255, alpha: 1)]
         let alphabetCharColor = [NSAttributedStringKey.foregroundColor: UIColor(red: 255/255, green: 255/255, blue: 255/255, alpha: 1)]
-        
+
         let attributedString = NSMutableAttributedString()
         for s in passwordString.unicodeScalars {
             let char: NSAttributedString
-            
```

Comment:
You should turn on "Xcode->Preferences->Text Editing -> Automatically trim trailing whitespace" to avoid these needless spaces all over. :)

Reply:
I actually have that on. I figured out whey it's there. Xcode added it because I performed a re-indent on the whole file. Not sure why.